cms/djangoapps/contentstore/tasks.py

```diff
@@ -242,13 +243,17 @@ def update_special_exams_and_publish(course_key_str):
     on_course_publish expects that the edx-proctoring subsystem has been refreshed
     before being executed, so both functions are called here synchronously.
     """
-    from cms.djangoapps.contentstore.proctoring import register_special_exams
+    from cms.djangoapps.contentstore.exams import register_exams
+    from cms.djangoapps.contentstore.proctoring import register_special_exams as register_exams_legacy
     from openedx.core.djangoapps.credit.signals import on_course_publish
 
     course_key = CourseKey.from_string(course_key_str)
     LOGGER.info('Attempting to register exams for course %s', course_key_str)
+
+    # Call the appropriate handler for either the exams IDA or the edx-proctoring plugin
+    register_exams_handler = register_exams if exams_ida_enabled(course_key) else register_exams_legacy
```

Comment:
I like the naming and structure here, it is very clear about what is what in very little space

Reply:
Ditto!